### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by quratulain25**
Login, Password Reset and Signup added.

### CODE REVIEW

File: LBM/settings.py [hunk 1]
```diff
@@ -120,3 +120,8 @@
 # https://docs.djangoproject.com/en/2.2/howto/static-files/
 
 STATIC_URL = '/static/'
+LOGIN_REDIRECT_URL = '/'
+LOGOUT_REDIRECT_URL = '/'
```
Comment (Ayub-Khan) [inline LBM/settings.py:124]: Logout redirect to login page

File: LBM/settings.py [hunk 2]
```diff
@@ -120,3 +120,8 @@
 # https://docs.djangoproject.com/en/2.2/howto/static-files/
 
 STATIC_URL = '/static/'
+LOGIN_REDIRECT_URL = '/'
```
Comment (Ayub-Khan) [inline LBM/settings.py:123]: Remove pyc files.

File: LBM/settings.py [hunk 3]
```diff
@@ -120,3 +120,8 @@
...
+LOGOUT_REDIRECT_URL = '/'
+
+EMAIL_BACKEND = "django.core.mail.backends.filebased.EmailBackend"
+EMAIL_FILE_PATH = os.path.join(BASE_DIR, "sent_emails")
```
Comment (Ayub-Khan) [inline LBM/settings.py:127]: Why email backend ?

File: LBM/urls.py [hunk 4]
```diff
@@ -15,8 +15,11 @@
...
 from django.contrib import admin
 from django.urls import path, include
+from django.views.generic import TemplateView
 
 urlpatterns = [
     path('admin/', admin.site.urls),
-    path(r'', include('management.urls')),
+    path('management/', include('management.urls')),
+    path('management/', include('django.contrib.auth.urls')),
+    path('', TemplateView.as_view(template_name='home.html'), name='home'),
```
Comment (Ayub-Khan) [inline LBM/urls.py]: Shift view based url to management

File: management/urls.py [hunk 5]
```diff
@@ -1,8 +1,11 @@
...
 from django.conf.urls import url
 from django.conf.urls.static import static
 
 from management import views
 
 urlpatterns = [
-    url(r'^$', views.index, name='index'),
+    # url(r'^$', views.index, name='index'),
+    # url(r'signup/', views.SignUP, name='signup'),
+    path('signup/', views.SignUp.as_view(), name='signup')
```
Comment (Ayub-Khan) [inline management/urls.py]: No need for signup

File: management/views.py [hunk 6]
```diff
@@ -1,6 +1,18 @@
 from django.shortcuts import render
 from django.http import HttpResponse
+from django.contrib.auth.forms import UserCreationForm
+from django.urls import reverse_lazy
+from django.views import generic
 
 # Create your views here.
-def index(request):
-    return HttpResponse('Hello World')
+from django.template import loader
```
Comment (Ayub-Khan) [inline management/views.py:2]: Move to imports

File: management/views.py [hunk 7]
```diff
@@ -1,6 +1,18 @@
...
+
+
+# def index(request):
```
Comment (Ayub-Khan) [inline management/views.py]: Remove comment

File: management/urls.py (hunk 5 above)
Comment (Ayub-Khan) [reply management/urls.py]: Remove all signup logic

File: management/views.py [hunk 8]
```diff
@@ -1,6 +1,18 @@
...
+#     template = loader.get_template('home.html')
+#     return HttpResponse(template.render({}, request))
+
+class SignUp(generic.CreateView):
+    form_class = UserCreationForm
+    success_url = reverse_lazy('login')
+    template_name = 'signup.html'
```
Comment (Ayub-Khan) [inline management/views.py]: Remove signup

**Review by Ayub-Khan — CHANGES_REQUESTED** [review]

File: LBM/settings.py (hunk 1 above)
Comment (quratulain25) [reply LBM/settings.py:124]: Shouldn't it redirect to home? As login option would be there?

File: LBM/urls.py [hunk 9]
```diff
@@ -15,11 +15,9 @@
 """
 from django.contrib import admin
 from django.urls import path, include
-from django.views.generic import TemplateView
 
 urlpatterns = [
     path('admin/', admin.site.urls),
     path('management/', include('management.urls')),
     path('management/', include('django.contrib.auth.urls')),
```
Comment (Ayub-Khan) [inline LBM/urls.py]: management ko accounts kar do

File: LBM/settings.py (hunk 1 above)
Comment (Ayub-Khan) [reply LBM/settings.py:124]: direct login is better

**Review by Ayub-Khan — APPROVED** [review]
1 NIT

**Pull request merged**